src/platform/packages/shared/kbn-esql-ast/src/mutate/commands/rerank/index.ts

```diff
@@ -34,15 +34,79 @@ export const list = (ast: ESQLAstQueryExpression): IterableIterator<ESQLAstReran
   ) as IterableIterator<ESQLAstRerankCommand>;
 };
 
+/**
+ * Sets or updates the query text for the RERANK command.
+ *
+ * @param cmd The RERANK command AST node to modify.
+ * @param query The query text to set.
+ */
 export const setQuery = (cmd: ESQLAstRerankCommand, query: string | ESQLStringLiteral) => {
-  if (typeof query === 'string') {
-    query = Builder.expression.literal.string(query);
+  const queryLiteral = typeof query === 'string' ? Builder.expression.literal.string(query) : query;
+  const firstArg = cmd.args[0];
+
+  cmd.query = queryLiteral;
+
+  if (
+    firstArg &&
+    !Array.isArray(firstArg) &&
+    firstArg.type === 'function' &&
+    firstArg.name === '='
+  ) {
+    // It's an assignment, update the right side of the expression
+    firstArg.args[1] = queryLiteral;
+  } else {
+    // It's a simple query, replace the first argument
+    cmd.args[0] = queryLiteral;
   }
+};
+
+/**
+ * Sets, updates, or removes the target field for the RERANK command.
+ * This refers to the `targetField =` portion of the command, which specifies                                  │
+ * the new column where the rerank score will be stored.                                                       │
+ *
+ * @param cmd The RERANK command AST node to modify.
+ * @param target The name of the target field to set. If `null` the assignment is removed.
+ */
+export const setTargetField = (cmd: ESQLAstRerankCommand, target: string | null) => {
```

Comment:
This function may be overkill, but it's flexible. Another option is to pass the key as an optional parameter to setQuery, and if it exists, overwrite what we have and manage them more simply. 1) no key, just query. 2) key, then do a key=value.